src/partition/src/splitter.rs

```diff
@@ -42,12 +43,35 @@ impl WriteSplitter {
         }
     }
 
-    pub fn split_insert(&self, insert: InsertRequest) -> Result<InsertRequestSplit> {
+    pub fn split_insert(
+        &self,
+        insert: InsertRequest,
+        schema: &Schema,
+    ) -> Result<InsertRequestSplit> {
         check_req(&insert)?;
-
-        let column_names = self.partition_rule.partition_columns();
-        let values = &insert.columns_values;
-        let partition_columns = find_partitioning_values(values, &column_names)?;
+        let mut insert = insert;
+        let partition_columns = self.partition_rule.partition_columns();
+        for column_schema in schema.column_schemas() {
+            let key = insert.columns_values.keys().next().unwrap();
+            let row_nums = insert.columns_values.get(key).unwrap().len();
```

Comment:
Is this will panic at where `schema.column_schemas().len()` != `insert.columns_values.keys().len()`?

Reply:
I guess that this doesn't panic because the following test passes:

https://github.com/GreptimeTeam/greptimedb/pull/1498/files#diff-34bcab2e99e5f158ebeceb2c917d7495ea0538ec0312ee1af1aaaf4c337dd887R302-R305

In the above test, `schema.column_schemas().len()` is not equal to `insert.columns_values.keys().len()`.